src/components/Api.js

```diff
@@ -0,0 +1,90 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  _handleRes = (res) => {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}`);
+  };
+
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, { headers: this._headers })
+      .then((res) => this._handleRes(res))
+      .then((data) => {
+        return data;
+      });
+  }
+
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, { headers: this._headers })
+      .then((res) => this._handleRes(res))
+      .then((data) => {
+        return data;
+      });
+  }
+
+  setUserInfo = ({ userName, userJob }) => {
+    return fetch(`${this._baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({
+        name: userName,
+        about: userJob,
+      }),
+    })
+      .then((res) => this._handleRes(res))
+      .then((data) => {
+        return data;
+      });
+  };
+
+  addNewCard = ({ cardName, cardLink }) => {
+    return fetch(`${this._baseUrl}/cards`, {
+      method: "POST",
+      headers: this._headers,
+      body: JSON.stringify({
+        name: cardName,
+        link: cardLink,
+      }),
+    })
+      .then((res) => this._handleRes(res))
+      .then((data) => {
+        return data;
+      });
+  };
+
+  deleteCard = (cardId) => {
+    return fetch(`${this._baseUrl}/cards/${cardId}`, {
+      method: "DELETE",
+      headers: this._headers,
+    }).then((res) => this._handleRes(res));
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._handleRes)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.

Reply:
I tried what you suggested, but I get an error while calling the api methods in index.js:
"index.js:44 Uncaught TypeError: Cannot read properties of undefined (reading 'then')"
is there something I need to change in the syntax? 
![image](https://github.com/user-attachments/assets/e9c7c753-dd6d-49ee-8b36-6015caae530b)
![image](https://github.com/user-attachments/assets/b4d88cdb-c348-4508-af1c-b762d7977585)

Reply:
> I tried what you suggested, but I get an error while calling the api methods in index.js: "index.js:44 Uncaught TypeError: Cannot read properties of undefined (reading 'then')" is there something I need to change in the syntax?  


You've forgotting `return` before `this._request `